src/android/Geolocation.java

```diff
@@ -185,14 +258,45 @@ private void addWatch(LocationContext locationContext) {
             request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
         }
 
-        requestLocationUpdatesIfSettingsSatisfied(locationContext, request);
+        requestLocationUpdatesIfSettingsSatisfiedGms(locationContext, request);
+    }
+
+    /**
+     * Add Watch Locations in devices which doesn't contains google services (Huawei Devices)
+     * @param locationContext the locationContext
+     */
+    private void addWatchHuawei(LocationContext locationContext) {
+        JSONArray args = locationContext.getExecuteArgs();
+        boolean enableHighAccuracy = args.optBoolean(1, false);
+        long maximumAge = args.optLong(2, 5000);
+
+        com.huawei.hms.location.LocationRequest request = com.huawei.hms.location.LocationRequest.create();
+
+        request.setInterval(maximumAge);
+
+        if(enableHighAccuracy) {
+            request.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+        }
+
+        requestLocationUpdatesIfSettingsSatisfiedHms(locationContext, request);
     }
 
     @SuppressLint("MissingPermission")
-    private void requestLocationUpdates(LocationContext locationContext, LocationRequest request) {
-        fusedLocationClient.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
+    private void requestLocationUpdatesGms(LocationContext locationContext, LocationRequest request) {
+        fusedLocationClientGms.requestLocationUpdates(request, locationContext.getLocationCallback(), null);
     }
 
+    @SuppressLint("MissingPermission")
+    private void requestLocationUpdatesHms(LocationContext locationContext, com.huawei.hms.location.LocationRequest request) {
+        if (fusedLocationProviderClientHms != null) {
```

Comment:
The `requestLocationUpdatesGms` doesn't have the equivalent null check. Shouldn't it be similar?

Also, if `fusedLocationProviderClientHms` is null, this method doesn't do anything as there's only code for the not null case. Does it make sense for the caller to call it if it doesn't do anything?